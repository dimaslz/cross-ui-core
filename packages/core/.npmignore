.DS_Store
.git
.npmrc