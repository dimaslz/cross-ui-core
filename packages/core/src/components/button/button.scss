.CrossUIButton {
  @apply cursor-pointer border-solid font-sans flex justify-center;

  @each $color in $colors {
    &.#{$color} {
      $color-value: map-get($colors-map, $color);

      @apply inline-flex items-center antialiased border-2 border-transparent leading-none text-sm px-4 bg-#{$color-value}-500 text-white font-semibold;
      @apply text-xs px-4;

      &.pill,
      &.circle {
        @apply rounded-full;
      }

      &.circle,
      &.square {
        @apply items-center justify-center p-1;
      }

      // SIZES
      &.small {
        @apply h-6 text-xs;

        &.circle,
        &.square {
          @apply w-6;
        }
      }

      &.medium {
        @apply h-8 text-sm;

        &.circle,
        &.square {
          @apply w-8;
        }
      }

      &.large {
        @apply h-10 text-base;

        &.circle,
        &.square {
          @apply w-10;
        }
      }

      &.simple {
        @apply outline-none bg-#{$color-value}-500;

        &:hover:not(:disabled) {
          @apply bg-#{$color-value}-600;
        }

        &:active:not(:disabled) {
          @apply bg-#{$color-value}-900;
        }
      }

      &.outline {
        @apply outline-none bg-transparent text-#{$color-value}-700 font-semibold border border-#{$color-value}-700;

        svg {
          @apply text-#{$color-value}-700;
        }

        &:hover:not(:disabled) {
          @apply border-#{$color-value}-400 text-#{$color-value}-400 border-transparent;
        }

        &:active:not(:disabled) {
          @apply text-#{$color-value}-900 border-#{$color-value}-900;
        }
      }

      &.simple, &.outline {
        &:focus {
          @apply ring ring-gray-200;
        }
      }
    }

    &:disabled {
      @apply opacity-50 cursor-not-allowed;
    }
  }

  &--full-width {
    @apply w-full;
  }
}
