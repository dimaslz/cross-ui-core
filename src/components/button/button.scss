@import '../../assets/css/_tailwind.scss';

$colors: 'blue', 'red', 'orange', 'green', 'teal', 'indigo', 'purple', 'pink';

.CrossUIButton {
  @apply cursor-pointer border-solid font-sans;

  @each $color in $colors {
    &.#{$color} {
      @apply inline-flex items-center antialiased border-2 border-transparent leading-none text-sm px-4 bg-#{$color}-500 text-white font-semibold;

      &.small,
      &.medium,
      &.large {
        @apply text-xs px-4;
        &.circle,
        &.square {
          @apply items-center justify-center p-1;
        }
      }
      &.small {
        @apply h-6 text-xs;
        &.circle,
        &.square {
          @apply w-6;
        }
      }

      &.medium {
        @apply h-8 text-sm;
        &.circle,
        &.square {
          @apply w-8;
        }
      }

      &.large {
        @apply h-10 text-base;
        &.circle,
        &.square {
          @apply w-10;
        }
      }

      &.simple {
        @apply outline-none bg-#{$color}-500;
        &:hover:not(:disabled) {
          @apply bg-#{$color}-600;
        }
        &:focus {
          @apply border-#{$color}-700;
        }
        &.pill,
        &.circle {
          @apply rounded-full;
        }
      }

      &.outline {
        @apply outline-none bg-transparent text-#{$color}-700 font-semibold border border-#{$color}-500;

        &:hover:not(:disabled) {
          @apply bg-#{$color}-600 text-white border-transparent;
        }
        &:focus {
          @apply border-#{$color}-700;
        }
        &.pill,
        &.circle {
          @apply rounded-full;
        }
      }

      &.elevated {
        @apply outline-none bg-white text-gray-800 font-semibold border border-gray-400 shadow;

        &:focus:not(:disabled),
        &:hover:not(:disabled) {
          @apply bg-gray-100;
        }
        &.pill,
        &.circle {
          @apply rounded-full;
        }
      }

      &:disabled {
        @apply opacity-50 cursor-not-allowed;

        &.pill,
        &.circle {
          @apply rounded-full;
        }
      }
    }
  }
}
