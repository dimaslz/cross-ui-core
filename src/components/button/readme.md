# tail-button



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description | Type      | Default    |
| ----------- | ------------ | ----------- | --------- | ---------- |
| `circle`    | `circle`     |             | `boolean` | `false`    |
| `color`     | `color`      |             | `string`  | `'blue'`   |
| `disabled`  | `disabled`   |             | `boolean` | `false`    |
| `fullWidth` | `full-width` |             | `boolean` | `false`    |
| `pill`      | `pill`       |             | `boolean` | `false`    |
| `selected`  | `selected`   |             | `boolean` | `false`    |
| `size`      | `size`       |             | `string`  | `'medium'` |
| `square`    | `square`     |             | `boolean` | `false`    |
| `type`      | `type`       |             | `string`  | `'button'` |
| `ui`        | `ui`         |             | `string`  | `'simple'` |


## Dependencies

### Used by

 - [app-home](../../../dev/components/home)

### Graph
```mermaid
graph TD;
  app-home --> cross-button
  style cross-button fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
