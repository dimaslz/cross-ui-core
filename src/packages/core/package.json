{
  "name": "@cross-ui/core",
  "version": "0.0.1",
  "description": "Cross UI, agnostic components based in Stencil",
  "main": "dist/index.cjs.js",
  "module": "dist/custom-elements/index.js",
  "es2015": "dist/esm/index.mjs",
  "es2017": "dist/esm/index.mjs",
  "types": "dist/custom-elements/index.d.ts",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "unpkg": "dist/cross-ui-core/cross-ui-core.js",
  "files": [
    "dist/",
    "loader/"
  ],
  "scripts": {
    "serve": "stencil build --dev --watch --serve",
    "start": "stencil build --dev --watch",
    "build": "rimraf dist && rimraf www && NODE_ENV=production stencil build --prod --docs"
  },
  "dependencies": {},
  "license": "MIT",
  "devDependencies": {}
}
