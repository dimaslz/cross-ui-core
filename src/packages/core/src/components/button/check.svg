<svg width="14" height="14" viewBox="0 0 14 14" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M4.15071 12.32C4.15071 12.2003 4.1276 12.1404 4.08138 12.1404L3.76248 12.2924C3.76248 12.2279 3.72551 12.1818 3.65156 12.1542L3.54064 12.1404C3.46669 12.1404 3.37426 12.1726 3.26334 12.2371C3.24485 12.1911 3.22174 12.145 3.19401 12.0989C3.16628 12.0529 3.14317 12.0114 3.12468 11.9746C3.00452 11.7443 2.88435 11.4909 2.76419 11.2146C2.65327 10.929 2.54697 10.6572 2.44529 10.3993C2.35285 10.1414 2.27891 9.93869 2.22345 9.79129C2.18647 9.67154 2.14488 9.4919 2.09866 9.25239C2.05244 9.01287 2.00622 8.70887 1.96001 8.34039C2.06168 8.40487 2.14025 8.43711 2.19572 8.43711C2.26042 8.43711 2.3205 8.34039 2.37596 8.14693C2.40369 8.18378 2.45453 8.2022 2.52848 8.2022C2.58394 8.2022 2.62554 8.18378 2.65327 8.14693L2.87511 7.81529L3.12468 7.8982H3.13855C3.15704 7.8982 3.17552 7.88899 3.19401 7.87057C3.2125 7.85214 3.24023 7.83372 3.2772 7.81529C3.35115 7.76923 3.40661 7.7462 3.44358 7.7462L3.48518 7.76002C3.71626 7.87057 3.86416 8.07323 3.92886 8.36802C4.09525 9.06814 4.26163 9.4182 4.42801 9.4182C4.59439 9.4182 4.78851 9.24317 5.01035 8.89311C5.12127 8.71808 5.23219 8.51542 5.34311 8.28511C5.46328 8.05481 5.58344 7.79687 5.70361 7.51129C5.7221 7.62184 5.74058 7.67711 5.75907 7.67711C5.80529 7.67711 5.88386 7.56196 5.99478 7.33166C6.11494 7.10136 6.30443 6.78354 6.56325 6.3782C6.71115 6.12948 6.89602 5.84851 7.11786 5.5353C7.34895 5.22208 7.5939 4.89966 7.85271 4.56802C8.11153 4.23639 8.36111 3.92317 8.60144 3.62839C8.85101 3.3336 9.07285 3.08026 9.26697 2.86839C9.46108 2.65651 9.60435 2.51833 9.69679 2.45384C10.048 2.21433 10.3253 1.98402 10.5287 1.76293C10.5195 1.82742 10.5056 1.8873 10.4871 1.94257C10.4779 1.98863 10.4732 2.02087 10.4732 2.0393C10.4732 2.07614 10.4917 2.09457 10.5287 2.09457L10.9169 1.90111V1.95639C10.9169 2.03008 10.9354 2.06693 10.9724 2.06693C11.0001 2.06693 11.0556 2.02548 11.1388 1.94257C11.222 1.85966 11.2682 1.79978 11.2774 1.76293L11.2497 1.95639L11.7211 1.68002L11.6102 1.92875C11.7581 1.82742 11.8644 1.77675 11.9291 1.77675C11.9661 1.77675 11.9938 1.79978 12.0123 1.84584C12.0308 1.88269 12.04 1.91954 12.04 1.95639C12.04 2.01166 12.0169 2.07614 11.9707 2.14984C11.9245 2.22354 11.8644 2.31105 11.7904 2.41239C11.735 2.48608 11.6425 2.59663 11.5131 2.74402C11.393 2.8822 11.2081 3.08948 10.9585 3.36584C10.7089 3.63299 10.3762 4.00608 9.96022 4.48511C9.8493 4.60487 9.6783 4.81675 9.44721 5.12075C9.21613 5.41554 8.95269 5.76099 8.6569 6.15711C8.37035 6.54402 8.0838 6.93554 7.79725 7.33166C7.51071 7.72778 7.25651 8.08705 7.03467 8.40948C6.81283 8.72269 6.65569 8.9576 6.56325 9.1142L5.70361 10.5651C5.51874 10.8783 5.36622 11.1363 5.24606 11.3389C5.12589 11.5324 5.03346 11.666 4.96875 11.7397C4.8301 11.9055 4.67758 12.0529 4.5112 12.1818L4.38642 12.1127L4.27549 12.1818L4.15071 12.32Z" fill="black"/>
</svg>
