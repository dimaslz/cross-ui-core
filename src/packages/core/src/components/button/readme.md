# tail-button



<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description | Type      | Default    |
| ----------- | ------------ | ----------- | --------- | ---------- |
| `circle`    | `circle`     |             | `boolean` | `false`    |
| `color`     | `color`      |             | `string`  | `'blue'`   |
| `disabled`  | `disabled`   |             | `boolean` | `false`    |
| `fullWidth` | `full-width` |             | `boolean` | `false`    |
| `pill`      | `pill`       |             | `boolean` | `false`    |
| `selected`  | `selected`   |             | `boolean` | `false`    |
| `size`      | `size`       |             | `string`  | `'medium'` |
| `square`    | `square`     |             | `boolean` | `false`    |
| `type`      | `type`       |             | `string`  | `'button'` |
| `ui`        | `ui`         |             | `string`  | `'simple'` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
