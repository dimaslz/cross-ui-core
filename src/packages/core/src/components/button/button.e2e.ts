import { newE2EPage } from '@stencil/core/testing';

describe('cross-button', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<cross-button></cross-button>');

    const element = await page.find('cross-button');
    expect(element).toHaveClass('hydrated');
  });
});

// describe('Check color property:', () => {
//   ['blue', 'red', 'orange', 'green', 'teal', 'indigo', 'purple', 'pink', 'yellow', 'gray'].forEach(color => {
//     it(`Color "${color}" is working`, async () => {
//       const page = await newE2EPage();
//       await page.setContent(`<cross-button color="${color}"></cross-button>`);

//       const button = await page.find('cross-button button');
//       const containsCorrectClasses = new RegExp(`(hover:)?bg-${color}-\\d{3}`, 'g').test(button.className);
//       expect(containsCorrectClasses).toBe(true);
//     });
//   });

//   it(`Can be disabled`, async () => {
//     const page = await newE2EPage();
//     await page.setContent(`<cross-button disabled="true"></cross-button>`);

//     const button = await page.find('cross-button button');
//     const containsCorrectClasses = new RegExp(`opacity-50 cursor-not-allowed`, 'g').test(button.className);
//     expect(containsCorrectClasses).toBe(true);
//   });
// })
