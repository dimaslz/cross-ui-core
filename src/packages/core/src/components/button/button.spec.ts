import { newSpecPage } from '@stencil/core/testing';
import { Button } from './button';

describe('cross-button', () => {
  it('builds', () => {
    expect(new Button()).toBeTruthy();
  });
  
  it('Snapshot', async () => {
    const page = await newSpecPage({
      components: [Button],
      html: `<cross-button>Click me!</cross-button>`
    });

    expect(page.root).toMatchSnapshot();
  });
});
