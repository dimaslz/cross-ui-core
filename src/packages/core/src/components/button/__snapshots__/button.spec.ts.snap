// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cross-button Snapshot 1`] = `
<cross-button>
  <mock:shadow-root>
    <button class="CrossUIButton blue false medium simple" type="button">
      <slot></slot>
    </button>
  </mock:shadow-root>
  Click me!
</cross-button>
`;
